src/explorer.api/Explorers/DiffixUtilities.cs

```diff
@@ -0,0 +1,42 @@
+namespace Explorer
+{
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+
+    using Explorer.Queries;
+
+    internal static class DiffixUtilities
+    {
+        internal static List<decimal> EstimateBucketResolutions(
+            long numSamples,
+            dynamic minSample,
+            dynamic maxSample,
+            long valuesPerBucketTarget)
+        {
+            Debug.Assert(numSamples > 0, "Argument numSamples should always be greater than zero.");
+
+            double valueDensity;
+            try
+            {
+                valueDensity = (double)numSamples / (maxSample - minSample);
+            }
+            catch (System.DivideByZeroException)
+            {
+                return new List<decimal> { 1M };
+            }
+
+            var bucketSizeEstimate = new BucketSize(valuesPerBucketTarget / valueDensity);
+
+            return (
+                from bucketSize in new List<BucketSize>
+                {
+                    bucketSizeEstimate.Smaller(steps: 2),
+                    bucketSizeEstimate,
+                    bucketSizeEstimate.Larger(steps: 2),
+                }
+                select bucketSize.SnappedSize)
+                .ToList();
```

Comment:
This could be simplified and optimized further:
```
            return new decimal[]
            {
                bucketSizeEstimate.Smaller(steps: 2).SnappedSize,
                bucketSizeEstimate.SnappedSize,
                bucketSizeEstimate.Larger(steps: 2).SnappedSize,
            };
```
Maybe it might be possible to also replace the return type `List<decimal>` with  `decimal[]`, to avoid the unnecessary wrapping.

Reply:
Good point, done: b2e912f58b5eb16f9221768ad4e2999251b52429